src/isar/state_machine/states/cancel.py

```diff
@@ -64,6 +69,21 @@ def start(self):
     def stop(self):
         self._log_state_transitions()
 
+    def _transform_results_to_asset_frame(
+        self, time_indexed_pose: Union[TimeIndexedPose, List[TimeIndexedPose]]
+    ):
```

Comment:
```suggestion
    def _transform_results_to_asset_frame(
        self, time_indexed_pose: TimeIndexedPose
    ):
```
Can we make this function only take a `TimeIndexedPose` and rather check if it is a single or a list where this function is called?

Reply:
Think the `Union[TimeIndexedPose, List[TimeIndexedPose]]` is the most accurate since we can get both based on the inspection type.